src/components/buttons.js

```diff
@@ -3,30 +3,33 @@ import classnames from 'classnames';
 import { SvgIcon } from './SvgIcon';
 
 /**
- * @typedef ButtonProps
+ * @typedef _ButtonProps
  * @prop {import('preact').Ref<HTMLButtonElement>} [buttonRef]
- * @prop {import("preact").ComponentChildren} [children]
- * @prop {string} [className]
  * @prop {string} [icon] - Name of `SvgIcon` to render in the button
  * @prop {'left'|'right'} [iconPosition] - Icon positioned to left or to
  *   right of button text
- * @prop {boolean} [disabled]
  * @prop {boolean} [expanded] - Is the element associated with this button
  *   expanded (set `aria-expanded`)
  * @prop {boolean} [pressed] - Is this button currently "active?" (set
  *   `aria-pressed`)
- * @prop {(event: Event) => void} [onClick]
  * @prop {'small'|'medium'|'large'} [size='medium'] - Relative button size:
  *   affects padding
  * @prop {Object} [style] - Optional inline styles
- * @prop {string} [title] - Button title; used for `aria-label` attribute
  * @prop {'normal'|'primary'|'light'|'dark'} [variant='normal'] - For styling: element variant
  */
 
+/**
+ * Fold in HTML button prop definitions into ButtonProps, but ignore `size` because it's inherited
+ * from HTMLElement and conflicts with the _ButtonProps.size prop above.
+ *
+ * @typedef {Omit<import('Preact').JSX.HTMLAttributes<HTMLInputElement>, 'size'> } HTMLInputElementProps
+ * @typedef {_ButtonProps & HTMLInputElementProps} ButtonProps
+ */
+
 /**
  * @typedef IconButtonBaseProps
  * @prop {string} icon - Icon is required for icon buttons
- * @prop {string} title - Title is required for icon buttons
+ * @prop {string} title - Button title; used for `aria-label` attribute
```

Comment:
I moved the description from the base to here since this is where it applies to now and we don't need to define this in the parent since its imported from HTMLButtonElement.  I left this prop here because its 1. required (unlike the its base prop) and 2. has special a function that's important to explain.